ehrapy/tools/supervised/_feature_importances.py

```diff
@@ -0,0 +1,157 @@
+        predicted_feature: The feature to predict by the model. Must be present in adata.var_names.
+        prediction_type: Whether the predicted feature is continuous or categorical. If the data type of the predicted feature
+            is not correct, conversion will be attempted.
+        model: The model to use for prediction. Choose between 'regression', 'svm', or 'rf'. Note that multi-class classification
+            is only possible with 'rf'. Defaults to 'regression'.
+        input_features: The features in adata.var to use for prediction. Should be a list of feature names. If 'all', all features
+            in adata.var will be used. Note that non-numeric input features will cause an error, so make sure to encode them properly
+            before. Defaults to 'all'.
+        layer: The layer in adata.layers to use for prediction. If None, adata.X will be used. Defaults to None.
+        test_split_size: The split of data used for testing the model. Should be a float between 0 and 1, representing the proportion.
+            Defaults to 0.2.
+        key_added: The key in adata.var to store the feature importances. Defaults to 'feature_importances'.
+        feature_scaling: The type of feature scaling to use for the input. Choose between 'standard', 'minmax', or None.
+            'standard' uses sklearn's StandardScaler, 'minmax' uses MinMaxScaler. Scaler will be fit and transformed
+            for each feature individually. Defaults to 'standard'.
+        percent_output: Set to True to output the feature importances as percentages. Note that information about positive or negative
+            coefficients for regression models will be lost. Defaults to False.
+        **kwargs: Additional keyword arguments to pass to the model. See the documentation of the respective model in scikit-learn for details.
+
+    Returns:
+        None
+    """
+    if predicted_feature not in adata.var_names:
+        raise ValueError(f"Feature {predicted_feature} not found in adata.var.")
+
+    if input_features != "all":
+        for feature in input_features:
+            if feature not in adata.var_names:
+                raise ValueError(f"Feature {feature} not found in adata.var.")
+
+    if model not in ["regression", "svm", "rf"]:
+        raise ValueError(f"Model {model} not recognized. Please choose either 'regression', 'svm', or 'rf'.")
+
+    if prediction_type not in ["continuous", "categorical"]:
+        raise ValueError(
+            f"Prediction type {prediction_type} not recognized. Please choose either 'continuous' or 'categorical'."
+        )
+
+    if feature_scaling not in ["standard", "minmax", None]:
+        raise ValueError(
+            f"Feature scaling type {feature_scaling} not recognized. Please choose either 'standard', 'minmax', or None."
+        )
+
+    if layer is not None:
+        data = adata.layers[layer].to_df()
+    else:
+        data = adata.to_df()
+
+    if prediction_type == "continuous":
```

Comment:
Since you're doing that check anyways, I wonder whether you can't just infer the `prediction_type` automatically?

Reply:
I could infer the `prediction_type`, but having it as a separate parameter serves as a kind of sanity check. For instance, if the variable we want to predict is saved numerically, it becomes unclear if the prediction should be continuous or categorically unless it was explicitly saved as a categorical variable. Since I assume that most features are encoded numerically (even when they are categorical), this would happen quite often. But, I also get that it would be nice to remove the parameter, so, if you feel that this double-check is unnecessary, I'll remove it and infer the `prediction_type`.

Reply:
I finally looked into this and implemented what you suggested, but I think we need to reconsider if this is really what we want. The main point is that if a feature is numerical, it shouldn't necessarily be considered continuous. For example, the `aline_flg` variable in the `mimic_2` dataset is set to either 0 or 1 for each patient. Hence, this should be treated as a classification problem, but as it is stored numerically, determining the `prediction_type` based on `adata.var["ehrapy_column_type"]`, will result in a continuous prediction. Changing this would be super annoying for the user, as they would need to manually change the encoding type in `adata.var["feature_encoding_type"]` or store values "yes" and "no" instead of 1 and 0.

I would suggest keeping the `prediction_type` parameter, but we could add the default option `auto`, where we infer the type using the approach you suggested. We log the inferred type, and if it's not what the user wants (as in the case of `aline_flg`), they still have the option to set the parameter to 'categorical'. Do you think adding the `auto` option just makes things more complex or would it be helpful?